futures-core/src/task/mod.rs

```diff
@@ -66,21 +82,21 @@ if_std! {
     impl<'a> Context<'a> {
         /// Create a new task context.
         ///
-        /// Task contexts are always equipped with:
+        /// Task contexts are equipped with:
         ///
         /// - Task-local data
         /// - A means of waking the task
         /// - A means of spawning new tasks, i.e. an [executor]()
         pub fn new(map: &'a mut LocalMap, waker: &'a Waker, executor: &'a mut Executor) -> Context<'a> {
-            Context { waker, map, executor }
+            Context { waker, map, executor: Some(executor) }
         }
 
         /// Get the default executor associated with this task.
         ///
         /// This method is useful primarily if you want to explicitly handle
         /// spawn failures.
         pub fn executor(&mut self) -> &mut Executor {
-            self.executor
+            *self.executor.as_mut().unwrap()
```

Comment:
Seems like we should make this `expect`, since users might actually encounter this in practice.